### DISCUSSION TITLE
How to combine node_export and icmp warnings in 1 rule?

### DISCUSSION THREAD

**Discussion by Thuan220820** [prometheus/prometheus#11754] [Q&A]
### Proposal

Hi team, I am currently overseeing the company's information system. I'm having a problem with the "virtual shutdown" warning. In this case, I have 2 rules as follows:
1. node_exporter: up == 0 --> labels.instance down (ip:port)
![image](https://user-images.githubusercontent.com/116623585/208851977-489c11c4-1f1a-496b-b627-77c14bf75fdd.png)

2. probe_success == 0 --> labels.instance down (ip)
![image](https://user-images.githubusercontent.com/116623585/208852050-608d43a7-aabc-4250-9c72-26dd82c3ac4a.png)

But each rule has its own shortcomings, for example node_exporter will be disabled when memory is high. How to combine these 2 rules into 1 complete rule?
Thank you team!

**Comment by JDTX0** [comment]
You should have a look at this post: https://www.robustperception.io/combining-alert-conditions/

Assuming your `instance` label values are equal on both `probe_success` and `up`, then this query should work:
```
up == 0 AND on(instance) probe_success{job=~"ICMP.*"} == 0
```

Keep in mind that this requires the `instance` values to be exactly identical. So if you have the ports in the values, e.g. `some-ip-or-hostname:9100` as instance (node_exporter) or `some-ip-or-hostname:9115` (blackbox exporter), then that query will never work.


In that case, you can do one of two things:
1) Use relabel_configs to strip the ports off of the series you're ingesting. See this [post](https://www.robustperception.io/controlling-the-instance-label/) on how to do that. Note: If you adjust your Prometheus configuration to relabel these values, it creates a new unique series as the value of the label differs. If that's undesirable for you, don't do this.
2) Replace the values at query-time using `label_replace`, as an example for your particular query:
```
label_replace(up == 0, "instance", "$1", "instance", ".*:9100") and on(instance) label_replace(probe_success{job=~"ICMP.*"} == 0, "instance", "$1", "instance", ".*:9115")
```

The `label_replace` query function strips away the port based on the regex provided, this allows both sides of the query to have matching label values and for the overall query to work.

You may also consider looking at [inhibition](https://prometheus.io/docs/alerting/latest/configuration/#inhibit_rule) rules. They're useful in cases where you want to silence an alert based another alert firing. for example, if ICMP isn't responding, you may wish to silence your node_exporter alert as the host is already known to be down.

Good luck!
